### ISSUE TITLE
A11y pain points in sidebar pattern

### ISSUE THREAD

**Issue by aviannamiller** [omeka/omeka-s#2339]
[Illogical focus order in sidebar pattern #2282](https://github.com/omeka/omeka-s/issues/2282) discusses illogical focus order issues in workflows that involve sidebars, both in moving from main content areas to sidebars and within sidebar patterns.

Compiling more general a11y issues with the sidebar in this space:

- As described more largely in https://github.com/omeka/omeka-s/issues/2331 but also applicable here, some `text`, `textarea` inputs and `select` elements on sidebars are missing labels.
- Semantics:
  - Sidebars are missing landmark roles. Assumedly, these should be contained within `aside` elements or otherwise tagged with `role=”complementary”`.
    - There may be instances of missing landmark roles within sidebar interfaces, as well, such as with buttons that exist for navigation purposes.
  - Sidebar titles are tagged as `<h3>`, while subheadings within sidebars are tagged with `<h4>`. My inclination is that the titles should be `<h2>` and subheadings `<h3>`, but there may be an argument for something else here.
- Pressing Enter in keyboard nav as a mode of selection causes the sidebar to close (in certain instances), depending where focus is held. This appears to be happening mostly when focus is in components of `select` inputs / dropdowns in sidebars, which more generally have their own focus issues (see https://github.com/omeka/omeka-s/issues/2335). This was first noticed in https://github.com/omeka/omeka-s/issues/2323.

**Comment by aviannamiller** [comment]
- Sidebar headings tend to only have 4.2:1 contrast with #676767 text on a #DFDFDF background. This should be at least 4.5:1.
